src/esp/bindings_js/modules/replay_viewer.js

```diff
@@ -0,0 +1,373 @@
+      } else {
+        stepSignedDist = action == "strafeLeft" ? stepDist : -stepDist;
+        adjustedRotation = -this.agentRotation - Math.PI * 1.0;
+      }
+      let state = this.sim.getAgentState();
+      state.position = [
+        state.position[0] + Math.cos(adjustedRotation) * stepSignedDist,
+        state.position[1],
+        state.position[2] + Math.sin(adjustedRotation) * stepSignedDist
+      ];
+      this.sim.setAgentState(state);
+    } else if (action == "turnLeft" || action == "turnRight") {
+      let rotDist = 0.035;
+      let rotSignedDist = action == "turnLeft" ? rotDist : -rotDist;
+      this.agentRotation += rotSignedDist;
+      this.syncAgentRotation();
+    } else if (action == "lookUp" || action == "lookDown") {
+      let rotDist = 0.015;
+      let MIN_LOOK_PITCH = -Math.PI * 0.2;
+      let MAX_LOOK_PITCH = Math.PI * 0.2;
+      let rotSignedDist = action == "lookUp" ? rotDist : -rotDist;
+      this.agentLookPitch += rotSignedDist;
+      this.agentLookPitch = Math.min(
+        Math.max(this.agentLookPitch, MIN_LOOK_PITCH),
+        MAX_LOOK_PITCH
+      );
+      this.syncAgentRotation();
+    }
+  }
+
+  handleKeypressUp(key) {
+    for (let a of this.actions) {
+      if (a.keyCode === key) {
+        this.removeActiveAction(a.name);
+        break;
+      }
+    }
+  }
+
+  handleKeypress(key, isRepeat) {
+    for (let a of this.actions) {
+      if (a.keyCode === key) {
+        const index = this.activeActions.indexOf(a.name);
+        if (index == -1) {
+          this.activeActions.push(a.name);
+        }
+        return true;
+      }
+    }
+
+    if (isRepeat) {
+      return false;
+    }
+
+    if (key == " ".charCodeAt(0)) {
+      this.isPaused = !this.isPaused;
+    } else if (key == "R".charCodeAt(0)) {
+      if (this.player) {
+        var currIndex = this.player.getKeyframeIndex();
+        var newIndex = Math.max(0, currIndex - 10);
+        this.player.setKeyframeIndex(newIndex);
+        this.isPaused = true; // pause after rewind
+      }
+    } else if (key == "L".charCodeAt(0)) {
+      this.components.logWrapper.style.display =
+        this.components.logWrapper.style.display == "block" ? "none" : "block";
+    }
+  }
+
+  bindKeys() {
+    var _self = this;
+    _self.keyBindListener = function(event) {
+      if (_self.handleKeypress(event.keyCode, event.repeat)) {
+        event.preventDefault();
+      }
+    };
+    document.addEventListener("keydown", _self.keyBindListener, true);
+
+    _self.keyBindListener2 = function(event) {
+      _self.handleKeypressUp(event.keyCode);
+    };
+    document.addEventListener("keyup", _self.keyBindListener2, true);
+  }
+}
+
+class ReplayViewer {
+  static preloadFiles(preloadFunc) {
+    let dataUrlBase = "";
+    let useLocalSourceFiles = window.location.href.indexOf("0.0.0.0") != -1;
+    if (useLocalSourceFiles) {
+      dataUrlBase = "data";
+    } else {
+      dataUrlBase = "https://www.ericundersander.com/habitat/data";
```

Comment:
Temporary hosting for these files

Reply:
I am really not a fan of hard coding this. Can't we force the user to specify it?

Reply:
>Can't we force the user to specify it?
I don't have a good answer here. I'm also not a fan of hard-coding this.

In practice, the "user" here is a Hab user looking to use replay_viewer.js as a reference to deploy their own replay. And they need to override not just the base URL but the full set of files (including all GLBs). This user can fork hab-sim and modify replay_viewer.js in place--they'll hard-code their specific set of files including specific URLs.

For now, I'll just remove the reference to ericundersander.com. Any user looking to deploy this will need to figure out their own URLs.